app/utils/auth0_parser.rb

```diff
@@ -0,0 +1,18 @@
+class Auth0Parser
+  def self.authenticate_request(request)
+    auth_token(request)
+  end
+
+  def self.http_token(request)
+    request.headers['Authorization'].split(' ').last
```

Comment:
I think this belongs to the Authentication class, so that it's only that class that needs to deal with understanding a request, which header the token is (Authentication), etc...
Parser should only see the token as a string, not the full request object.
It'll also allow you to test it.

Reply:
That makes sense. When I move this to the Authentication class, I'd probably want to put it in a private method though - so how would I test that?

Reply:
I'll let your existing tests guide you :)
Think about what'd happen in your tests if you used this header format as an input of the "only" public method that the class has.

Reply:
Changing that felt like a pure refactor, since I'm just moving a private method from one class to another. I totally agree that it belongs in Authentication. Changing it kept all tests passing, so I'm not sure what test to add now that I've moved it, but maybe I'm missing something.

Reply:
Tests are passing because they already have the expected format, which is everything that you need.
You know that you're reading the header correctly because your mock parser only returns something under a very specific input (the token part of the header).
If you were to break how the header gets read, like let's say you forget to do the `.split(" ").last` your tests would fail.

Reply:
Yup! I had already changed the tests so the parser was being given something that looks more like an actual header, so everything still passes.